platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/internal/cc/impl/problems/ConfigurationCacheProblems.kt

```diff
@@ -411,4 +438,17 @@ class ConfigurationCacheProblems(
             else -> "$this $plural"
         }
     }
+
+    private abstract class ErrorsAreProblemsProblemsListener(
```

Comment:
:nail_care: Given that the only difference in the implementations is the severity of the produced problem, this class can accept it in the constructor and become non-abstract.